crates/core/src/factories/core_fetch/test/outgoing_request_queue.rs

```diff
@@ -273,200 +269,170 @@ async fn filter_requests_for_held_ops() {
         fetch,
-        requests_sent,
+        peer_meta_store,
         _transport,
         ..
-    } = setup_test(&config, true).await;
+    } = setup_test(&config).await;
 
     let op_list_1 = create_op_id_list(2);
-    let peer_url_1 = random_peer_url();
+    let unresponsive_url = random_peer_url();
 
     // Create a second agent to later check that their ops have not been removed.
     let op_list_2 = create_op_id_list(2);
-    let peer_url_2 = random_peer_url();
-
-    fetch
-        .request_ops(op_list_1.clone(), peer_url_1.clone())
-        .await
-        .unwrap();
-
-    // Add control agent's ops to set.
-    fetch
-        .request_ops(op_list_2, peer_url_2.clone())
+    let responsive_url = random_peer_url();
+
+    peer_meta_store
+        .set_unresponsive(
+            unresponsive_url.clone(),
+            Timestamp::now(),
+            Timestamp::now(),
+        )
         .await
         .unwrap();
 
     // Check that when the requests are dropped, the queue is treated as drained.
     let (tx, rx) = futures::channel::oneshot::channel();
     fetch.notify_on_drained(tx);
 
-    // Back off agent the maximum possible number of times.
-    let last_back_off_interval = {
-        let mut lock = fetch.state.lock().unwrap();
-        assert!(op_list_1.iter().all(|op_id| {
-            lock.requests.contains(&(op_id.clone(), peer_url_1.clone()))
-        }));
-        for _ in 0..config.num_back_off_intervals + 1 {
-            lock.back_off_list.back_off_peer(&peer_url_1);
-        }
-
-        lock.back_off_list.last_back_off_interval_ms
-    };
-
-    // Wait for back off interval to expire. Afterwards the request should fail again and all
-    // the agent's requests should be removed from the set.
-    tokio::time::sleep(Duration::from_millis(last_back_off_interval as u64))
-        .await;
+    fetch
+        .request_ops(op_list_1.clone(), unresponsive_url.clone())
+        .await
+        .unwrap();
 
+    // All ops in the request set are to be sent to the unresponsive URL.
     assert!(fetch
         .state
         .lock()
         .unwrap()
-        .back_off_list
-        .has_last_back_off_expired(&peer_url_1));
-
-    let current_number_of_requests_to_agent_1 = requests_sent
-        .lock()
-        .unwrap()
+        .requests
         .iter()
-        .filter(|(_, a)| *a != peer_url_1)
-        .count();
+        .all(|(_, peer_url)| *peer_url == unresponsive_url));
 
-    // Wait for another request attempt to agent 1, which should remove all of their requests
+    // Check that the drop has happened.
+    tokio::time::timeout(Duration::from_secs(1), rx)
+        .await
+        .expect("Timed out waiting for drained notification")
+        .unwrap();
+
+    // Add control agent's ops to set.
+    fetch
+        .request_ops(op_list_2, responsive_url.clone())
+        .await
+        .unwrap();
+
+    // Wait for a request attempt to the unresponsive URL, which should remove all of its requests
```

Comment:
I'm not clear what this check is testing here, it seems redundant. Or at least sending another message to the responsive url above seems redundant. Since `notify_on_drained` had been emitted earlier I would naturally expect this `iter_check!` here to return immediately since requests of the unresponsive peer had already been filtered out. Or am I missing something?

Reply:
No, you're not missing something. The refactor of this existing test was incomplete. I could have just checked for the drained queue and not check the other part. I've changed it so that there's the control agent to ensure that only the unresponsive agent's requests are dropped.

Reply:
https://github.com/holochain/kitsune2/pull/237/commits/22f1116d641015c15d94fe6456332f004d559662